File: 221121/LongJump.java
public class LongJump {
    public long solution(int n) {
        int[] cases = new int[2001];

        cases[1] = 1;
        cases[2] = 2;
        for (int i = 3; i <= 2000; i += 1) {
            cases[i] = (cases[i - 2] + cases[i - 1]) % 1234567;
        }

        return cases[n];
    }
}

File: solutions/Previous/221122/LongJump.js
export default function solution(n) {
  const initialCase = Array(3).fill(0).map((_, index) => index);

  const allCases = Array(n + 1).fill(0).reduce((cases, _, index) => {
    if (index <= 2) {
      return cases;
    }
    const casesAdded = [...cases,
      (cases[index - 2] + cases[index - 1]) % 1234567];
    return casesAdded;
  }, initialCase);

  return allCases[n];
}
